src/saltext/mysql/cache/mysql_cache.py

```diff
@@ -303,23 +248,26 @@ def flush(bank, key=None):
         data = (bank,)
     else:
         data = (bank, key)
-        query += " AND etcd_key=%s"
+        query += " AND cache_key=%s"
 
     cur, _ = run_query(__context__.get("mysql_client"), query, args=data)
     cur.close()
 
 
-def ls(bank):
+def list_(bank):
     """
-    Return an iterable object containing all entries stored in the specified
-    bank.
+    Return an iterable object containing all entries stored in the specified bank.
     """
     _init_client()
-    query = "SELECT etcd_key FROM {} WHERE bank=%s".format(__context__["mysql_table_name"])
-    cur, _ = run_query(__context__.get("mysql_client"), query, args=(bank,))
+    query = "SELECT bank FROM {}".format(__context__["mysql_table_name"])
```

Comment:
issue(blocking): This does not list the requested cache `bank`, it lists all. See general review comment for details.

Reply:
I agree, I made a mistake. The cache won't only hold minions cache...